### PR TITLE
Side panel

### LINKED ISSUE #68: Show side panel for admins
If the logged-in user is admin, then show them different layout with side panel. For now, side pannel can contain only search visitors page

### PR DESCRIPTION
**Pull request by sangeethailango**
Fixed #68

### CODE REVIEW

File: lib/mmbooking_Clone/user/visitor.ex [hunk 1]
```diff
@@ -28,6 +28,6 @@ defmodule Mmbooking_Clone.User.Visitor do
   def changeset(visitor, attrs) do
     visitor
     |> cast(attrs, [:id, :email_id, :first_name, :last_name, :dob, :country, :city, :have_you_visited_inner_chamber, :last_date_of_visit, :preferred_date, :alternate_date_of_visit, :place_of_stay, :arrival_date, :departure_date, :notes])
-    |> validate_required([:email_id, :first_name, :last_name, :dob, :country, :city, :have_you_visited_inner_chamber,  :preferred_date, :alternate_date_of_visit, :place_of_stay, :arrival_date, :departure_date, :notes])
```
Comment (sakthibalan15) [inline lib/mmbooking_Clone/user/visitor.ex:31]: Why you removed fields from validate required?

File: lib/mmbooking_Clone_web/components/layouts/root.html.heex [hunk 2]
```diff
@@ -39,20 +39,66 @@
...
-            href={~p"/log_in"}
-            class="text-[0.8125rem] leading-6 text-zinc-900 font-semibold hover:text-zinc-700"
           >
             Log in
           </.link>
         </li>
       <% end %>
     </ul>
-    <%= @inner_content %>
+    <%= if @current_user do %>
```
Comment (sakthibalan15) [inline lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: Why sidebar coming in root.html. What about admin layout?

File: lib/mmbooking_Clone/user/visitor.ex (hunk 1 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone/user/visitor.ex:31]: For the Add Visitor form, I need to insert only upto 6 to 7 fields, I don't need to insert other fields so i removed other fields that are not required.

File: lib/mmbooking_Clone_web/components/layouts/root.html.heex (hunk 2 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: What do you mean by admin layout? Admin can only see the sidebar. normal users cannot see this sidebar. See the below picture. 
[

](url)

File: lib/mmbooking_Clone/user/visitor.ex (hunk 1 above)
Comment (sakthibalan15) [reply lib/mmbooking_Clone/user/visitor.ex:31]: What about new visitor registration. Is this same changeset you using for new registration right?

File: lib/mmbooking_Clone_web/components/layouts/root.html.heex (hunk 2 above)
Comment (sakthibalan15) [reply lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: Don't show menus in welcome page, visitors form page and Visitior home page. The menus only visible for admins not for visitors.

File: lib/mmbooking_Clone/user/visitor.ex (hunk 1 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone/user/visitor.ex:31]: But there, we are only saving into the database only after every field is entered. We are saving only on the verify details page only.

File: lib/mmbooking_Clone_web/components/layouts/root.html.heex (hunk 2 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: @sakthibalan15 Do i need to change anything

Comment (sangeethailango) [reply lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: I have used root.html just like how we used to write code for navbars.

File: lib/mmbooking_Clone/user/visitor.ex (hunk 1 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone/user/visitor.ex:31]: @sakthibalan15 tole me to use different changeset

File: lib/mmbooking_Clone_web/components/layouts/root.html.heex (hunk 2 above)
Comment (sangeethailango) [reply lib/mmbooking_Clone_web/components/layouts/root.html.heex:48]: @sakthibalan15 told me to search about this.

**Pull request merged**